bigtable/google/cloud/bigtable/table.py

```diff
@@ -312,18 +317,25 @@ def mutate_rows(self, rows):
                   corresponding to success or failure of each row mutation
                   sent. These will be in the same order as the `rows`.
         """
-        mutate_rows_request = _mutate_rows_request(self.name, rows)
-        client = self._instance._client
-        responses = client._data_stub.MutateRows(mutate_rows_request)
-
-        responses_statuses = [
-            None for _ in six.moves.xrange(len(mutate_rows_request.entries))]
-        for response in responses:
-            for entry in response.entries:
-                responses_statuses[entry.index] = entry.status
-                if entry.status.code == 0:
-                    rows[entry.index].clear()
-        return responses_statuses
+        delay_millis = 1000
+        delay_mult = 1.5
+        max_delay_millis = 15 * 1000
+        total_timeout_millis = 5 * 60 * 1000
+
+        now = time.time()
+        deadline = now + total_timeout_millis / _MILLIS_PER_SECOND
```

Comment:
Currently, I don't think it matters, but if the retry params become user configureable, sub-millis precision will be lost here due to integer division.

Reply:
Changed to a float division in https://github.com/GoogleCloudPlatform/google-cloud-python/pull/4116/commits/28366582be449fe77115344abf10c81a2aa21ea1